cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -556,3 +692,29 @@ def _new_filter(self, pattern=None, param=None):
         if param is None:
             param = "@logStream"
         return _Filter(pattern, param)
+
+    def _add_error_metrics_graph_widgets(self, cluster_common_errors):
+        """Add cluster error metrics graph widgets."""
+        custom_namespace = "ParallelCluster/Health/" + self.config.cluster_name
+        widgets_list = []
+        for error in cluster_common_errors:
+            metric_list = []
+            for new_filter in error.metric_filters:
+                self.custom_pcluster_metric_filter(
+                    metric_name=new_filter.metric_name,
+                    filter_pattern=new_filter.filter_pattern,
+                    custom_namespace=custom_namespace,
+                    metric_value=new_filter.metric_value,
+                )
```

Comment:
Just a question out of curiosity:
What does this `self.custom_pcluster_metric_filter()` step do?
I see the `custom_pcluster_metric_filter()` function returns a metric_filter, but here the returned metric filter does not get stored or used right? And it's also not an attribute of any object? How does it influence the `cloudwatch_metric` or `metric_list`?
But the way, this `custom_pcluster_metric_filter()` seems to be a method, should the name contain a verb, like `create_custom_pcluster_metric_filter()`?

Reply:
Good question, the custom_pcluster_metric_filter here create a metric filter cfn resource with the log group. We do not return or used it because don't need to use any attribute of the metric_filter here. Metrics are generated with namespace, metric_name and dimension specified in metrics filters. 
Yes, we can rename the function.